src/IO/S3/Client.cpp

```diff
@@ -599,7 +601,9 @@ Client::doRequestWithRetryNetworkErrors(const RequestType & request, RequestFn r
                 last_exception = std::current_exception();
 
                 auto error = Aws::Client::AWSError<Aws::Client::CoreErrors>(Aws::Client::CoreErrors::NETWORK_CONNECTION, /*retry*/ true);
-                client_configuration.retryStrategy->CalculateDelayBeforeNextRetry(error, attempt_no);
+                auto sleep_ms = client_configuration.retryStrategy->CalculateDelayBeforeNextRetry(error, attempt_no);
+                LOG_WARNING(log, "Request failed, now waiting {} ms before attempting again", sleep_ms);
+                sleepForMilliseconds(sleep_ms);
```

Comment:
The retrying code for `ConnectionResetException` was here already however with pauses between attempts it must be more reliable.

Reply:
I want to say an opposite opinion.
That there is no retry it is a bug, my actually. I have mention it recently.

But I think that we should do no sleep here. Look, the http_client do a good sleep call. It sleeps on a conditional variable for be able to cancel sleep if the server is going to shutdown.
https://github.com/capeanalytics/aws-sdk-cpp/blob/e88f75add5a9433601b6d46fe738e493da56ac3b/aws-cpp-sdk-core/source/http/HttpClient.cpp#L45-L48

If we do blind sleep we hung the thread for that time. Since it is retry of other retries, I propose do not sleep here. That retries are rare.

Reply:
Or lets sleep really not long here, what ever CalculateDelayBeforeNextRetry returns. Lest say not more 0.5s.

Reply:
Also it will be good to have some profile events for this.